modules/phase_field/src/materials/MultiBarrierFunctionMaterial.C

```diff
@@ -45,6 +47,14 @@ MultiBarrierFunctionMaterial::computeQpProperties()
   for (unsigned int i = 0; i < _num_eta; ++i)
   {
     const Real n = (*_eta[i])[_qp];
+
+    if (_well_only && n >= 0.0 && n <= 1.0) {
+      _prop_g[_qp] = 0.0;
+      (*_prop_dg[i])[_qp] = 0.0;
+      (*_prop_d2g[i])[_qp] = 0.0;
+      return;
```

Comment:
I think this is the wrong logic, you are going to return after the first time you enter this loop, and therefore won't zero out all `_num_eta` entries in `_prop_dg` and `_prog_d2g`.